File: PAT/basic_level/1054_求平均值/average.py
from __future__ import print_function

def isInteger(str_num):
	if str_num[0].isdigit() or str_num[0]=='-':
		for i in str_num[1:]:
			if not i.isdigit():
				return 0
		return 1
	else:
		return 0

def isFloat(str_num):
	return str_num.count('.')==1

us_N = int(raw_input())
str_num = raw_input().split(' ')
sigma = 0.0
cnt = 0

for i in range(us_N):
	n = str_num[i]
	if isInteger(n):
		n = int(n)
		if n>=-1000 and n<=1000:
			sigma += n
			cnt += 1
			continue
	elif isFloat(n):
		n1, n2 = n.split('.')
		if len(n2)<=2:
			n = float(n)
			if n>=-1000 and n<=1000:
				sigma += n
				cnt += 1
				continue
	print("ERROR:", n, "is not a legal number")

if cnt>0:
	if cnt==1:
		print('The average of', cnt, 'number is %.2f' % sigma)
	else:
		print('The average of', cnt, 'numbers is %.2f' % (sigma/cnt))
else:
	print('The average of 0 numbers is Undefined')


'''
	# inte, flt = str(sigma/cnt).split('.')
	# if len(flt)<2:
	# 	flt += '0'
	# print('The average of', cnt, 'numbers is %s.%s' %(inte, flt[0:2]))

7
5 -3.2 aaa 9999 2.3.4 7.123 2.35

2
aaa -9999

2
0 999.99

2
0 -999.99

1
0

2
0 1000.00

1
-1000

1
12a2

2
-999.99 1000.00

2
-999.99 -1000.00





'number[s]'


'-10'.isdigit() is False
isdigit() judges str[0], that is '-'

'''

File: PAT/basic_level/1054_求平均值/average.cpp
#include <cstdio>
#include <cstring>
using namespace std;

int main(){
	int us_N, i=0, j=0, cnt=0, is_equal=1;
	char str_num[100]={0}, str_in[100]={0};
	double sigma=0.0, num=0.0;

	scanf("%d", &us_N);
	for( i=0; i<us_N; i++ ){
		scanf("%s", str_in);
		sscanf(str_in, "%lf", &num);
		sprintf(str_num, "%.2lf", num);
		is_equal = 1;
		for( j=0; j<strlen(str_in); j++ ){
			if( str_in[j]!=str_num[j] )
				is_equal = 0;
		}
		if( is_equal==0 || num<-1000 || num>1000 ){
			printf("ERROR: %s is not a legal number\n", str_in);
		}else{
			sigma += num;
			cnt++;
		}
	}

	if( cnt>1 )
		printf("The average of %d numbers is %.2lf", cnt, sigma/cnt);
	else if( cnt==1 )
		printf("The average of 1 number is %.2lf", sigma);
	else
		printf("The average of 0 numbers is Undefined");

	return 0;
}
